File: WebMonitor/ClientApp/src/model/file-information.ts
/**
 * File information response from the server.
 */
export interface FileInformationResponse {
    name: string;
    path: string;
    size: bigint;
    created: Date | string;
    lastModified: Date | string;
    lastAccessed: Date | string;
}

/**
 * File information with dates converted to Date objects.
 */
export interface FileInformation extends FileInformationResponse {
    created: Date;
    lastModified: Date;
    lastAccessed: Date;
}

File: WebMonitor/Model/FileInformation.cs
namespace WebMonitor.Model;

public class FileInformation
{
    public string Name { get; }
    public string Path { get; }
    public ulong Size { get; }
    public DateTime LastModified { get; }
    public DateTime LastAccessed { get; }
    public DateTime Created { get; }

    public FileInformation(FileInfo fileInfo)
    {
        Name = fileInfo.Name;
        Path = fileInfo.FullName;
        Size = (ulong)fileInfo.Length;
        // Times are in UTC
        LastModified = fileInfo.LastWriteTimeUtc;
        LastAccessed = fileInfo.LastAccessTimeUtc;
        Created = fileInfo.CreationTimeUtc;
    }
}
